### DISCUSSION TITLE
What are the best practices when distributing a custom listing plugin?

### DISCUSSION THREAD

**Discussion by coatless** [quarto-dev/quarto-cli#11209] [Q&A]
### Description

I'm working on modifying some listing behavior following from [Custom Listings](https://quarto.org/docs/websites/website-listings-custom.html). 

Reading through [Extensions](https://quarto.org/docs/extensions/), I'm not seeing a nice distribution mechanism similar to RevealJS plugins or a custom metadata bundle. Is there a good approach for sharing a custom listing?

For example, if I'm linking research output, my custom listing would be:

- papers.yml
- template.ejs
- Qmd index

I can either distribute the work by:

1. Zipping it and placing it somewhere for folks to download
2. Referencing a folder in a git repo
3. Blog post with copy code sections

But, these seem ill-suited for getting folks up and going.

**Comment by mcanouil** [comment]
I think it would be a feature request to extend the extension mechanism to have `quarto create extension listing` or something like that.

**Reply by coatless** [reply]
One side question: For the `ejs` template is there a reason why we need to nest it in a raw HTML cell despite the file being an `ejs`? Shouldn't it be evaluated as-is? e.g.

````md
```{=html}
// EJS here
```
````

vs.

````md
// EJS here
````

**Reply by mcanouil** [reply]
It's because (I think) you can use all Quarto markdown features as is in it.

- https://github.com/mcanouil/quarto-issues-experiments/blob/main/test-quarto-gallery/gallery.ejs
- https://github.com/mcanouil/quarto-extensions/blob/main/assets/ejs/authors.ejs

**Reply by cscheid** [reply]
It's because these ejs templates produce _markdown_, and not HTML. The expansion of EJS templates for listings happens in https://github.com/quarto-dev/quarto-cli/blob/main/src/project/types/website/listing/website-listing-template.ts#L57
